pkg/operatorsource/catalogsourceconfigcrud_test.go

```diff
@@ -0,0 +1,175 @@
+package operatorsource_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis"
+	"github.com/operator-framework/operator-marketplace/pkg/operatorsource"
+	"github.com/operator-framework/operator-marketplace/pkg/phase"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// Use Case: CatalogSourceConfig object has not been created yet.
+// Expected Result: A properly populated CatalogSourceConfig should get created
+func TestCRUD_Create_NewCatalogSourceConfig(t *testing.T) {
+
+	// Set up new fakeclient
+	scheme := runtime.NewScheme()
+	apis.AddToScheme(scheme)
+
+	fakeclient := fake.NewFakeClientWithScheme(scheme)
+
+	namespace, name := "marketplace", "foo"
+	context := context.TODO()
+	opsrc := helperNewOperatorSourceWithEndpoint(namespace, name, "remote")
+	manifests := "a,b,c"
+	catalogsourceconfigcrud := operatorsource.NewCatalogSourceConfigCRUD(fakeclient)
+
+	err := catalogsourceconfigcrud.CreateCatalogSourceConfig(context, opsrc, manifests)
+
+	assert.NoError(t, err)
+}
```

Comment:
yeah, the assertion needs to `get` the object created from the fake client and assert that it is `equal` to the expected. This is more like `state` verification as opposed to `behavior` verification with a mock.